ui/revenuecatui/src/main/kotlin/com/revenuecat/purchases/ui/revenuecatui/helpers/OfferingToStateMapper.kt

```diff
@@ -50,18 +51,37 @@ internal fun Offering.validatedPaywall(
 
 @Suppress("ReturnCount")
 private fun PaywallData.validate(): Result<PaywallTemplate> {
-    val (_, localizedConfiguration) = localizedConfiguration
+    val template = validateTemplate() ?: return Result.failure(PaywallValidationError.InvalidTemplate(templateName))
 
-    val invalidVariablesError = localizedConfiguration.validateVariables()
-    if (invalidVariablesError != null) {
-        return Result.failure(invalidVariablesError)
-    }
+    when (template.configurationType) {
+        PackageConfigurationType.SINGLE, PackageConfigurationType.MULTIPLE -> {
+            val (_, localizedConfiguration) = localizedConfiguration
 
-    val template = validateTemplate() ?: return Result.failure(PaywallValidationError.InvalidTemplate(templateName))
+            val invalidVariablesError = localizedConfiguration.validateVariables()
+            if (invalidVariablesError != null) {
+                return Result.failure(invalidVariablesError)
+            }
 
-    val invalidIconsError = localizedConfiguration.validateIcons()
-    if (invalidIconsError != null) {
-        return Result.failure(invalidIconsError)
+            val invalidIconsError = localizedConfiguration.validateIcons()
+            if (invalidIconsError != null) {
+                return Result.failure(invalidIconsError)
+            }
+        }
+        PackageConfigurationType.MULTITIER -> {
+            val (_, localizedConfigurationByTier) = tieredLocalizedConfiguration
+
+            localizedConfigurationByTier.entries.forEach { (_, localizedConfiguration) ->
```

Comment:
I probably need to validate tier count here 🤔

Reply:
So validate that there is at least 2+ tiers? Not sure if we could allow 1... Doesn't make much sense, but shouldn't break anything so 🤷